src/wallet/spend.h

```diff
@@ -224,7 +224,7 @@ util::Result<CreatedTransactionResult> CreateTransaction(CWallet& wallet, const
  * Insert additional inputs into the transaction by
  * calling CreateTransaction();
  */
-util::Result<CreatedTransactionResult> FundTransaction(CWallet& wallet, const CMutableTransaction& tx, std::optional<unsigned int> change_pos, bool lockUnspents, const std::set<int>& setSubtractFeeFromOutputs, CCoinControl);
+util::Result<CreatedTransactionResult> FundTransaction(CWallet& wallet, const CMutableTransaction& tx, const std::vector<CRecipient>& recipients, std::optional<unsigned int> change_pos, bool lockUnspents, CCoinControl);
```

Comment:
Not a huge fan of this change. Now if we pass tx.vout they'll be ignored.

I'd prefer if we pass `txin` and `recipients` and don't pass `tx` at all. That would be a clearer interface in my opinion.

Reply:
I agree, see my comment in the description. Ultimately, I think `FundTransaction` should take a set of inputs and a set of outputs and return a `CreatedTransactionResult`. #25273 and this PR get us incrementally closer to that, but I don't want to fully tackle that in this PR as it's a bit more complicated than it seems, and the primary goal of this PR is to create the `CRecipients` objects in a standard way across all the RPC calls and pass them to `FundTransaction`

Reply:
I think it would be useful to assert that `tx.vout.empty()` somewhere to make it clear that it is not used. Perhaps this function can do `tx.vout.clear()`, and the `FundTransaction` called by it can do the check.

Reply:
Good point; I'll add a check and a comment.

Reply:
Rather than clearing tx.vout in the outer RPC function call, I am clearing tx.vout at the call sites of FundTransaction. Otherwise, I would have needed to make `tx` non-const, which seems risky since we are still using it to pass `tx.vins`